pyartcd/pyartcd/pipelines/update_golang.py

```diff
@@ -251,6 +238,42 @@ async def run(self):
             dry_run=self.dry_run,
         )
 
+    def regen_repo_with_complete(self, el_v, timeout):
+        build_tag = f'rhaos-{self.ocp_version}-rhel-{el_v}-build'
+        task_id = self.koji_session.newRepo(build_tag)
+        _LOGGER.info(f"Regenerating repo for tag: {build_tag} with task {task_id}, waiting max to 10 mins...")
+        # regen repo usually take a few mins, wait max to 10 mins
+        wait_min = 0
+        while wait_min < timeout:
+            time.sleep(60)
+            gen_repo_task_info = self.koji_session.getTaskInfo(task_id)
+            if gen_repo_task_info['completion_time']:
+                _LOGGER.info(f"Regenerating repo for tag: {build_tag} complete.")
+                return True
+            _LOGGER.info(f"Regenerating repo for tag: {build_tag} task {task_id} is still running")
+            wait_min += 1
+        return False
+
+    def tag_builds_override(self, el_v, nvr):
+        if not self.koji_session.logged_in:
+            _LOGGER.info("user logged out from session, login again")
+            self.koji_session.gssapi_login()
+        if el_v == 8:
+            rhel8_module_tag = self.get_module_tag(nvr, el_v)
+            if rhel8_module_tag:
+                latest_rhel8_builds = self.koji_session.listTagged(rhel8_module_tag, latest=True, inherit=True)
+                # need to tag delve go-toolset golang 3 module builds
+                for el8_nvr in [b['nvr'] for b in latest_rhel8_builds]:
+                    self.koji_session.tagBuild(f'rhaos-{self.ocp_version}-rhel-{el_v}-override', el8_nvr)
+                    _LOGGER.info(f"Tagged {el8_nvr} with rhaos-{self.ocp_version}-rhel-{el_v}-override tag")
+            else:
+                # if a rhel8 golang is not an module then treat it as rpm
+                self.koji_session.tagBuild(f'rhaos-{self.ocp_version}-rhel-{el_v}-override', nvr)
+                _LOGGER.info(f"Tagged {nvr} with rhaos-{self.ocp_version}-rhel-{el_v}-override tag")
+        elif el_v == 9:
+            self.koji_session.tagBuild(f'rhaos-{self.ocp_version}-rhel-{el_v}-override', nvr)
+            _LOGGER.info(f"Tagged {nvr} with rhaos-{self.ocp_version}-rhel-{el_v}-override tag")
+
```

Comment:
It would be good to deduplicate this. A var `nvrs_to_tag` which is populated in the if-else branch, and then they are tagged in the `tag`